raft/server.cc

```diff
@@ -1344,44 +1344,44 @@ future<> server_impl::applier_fiber() {
-
-               // It may happen that _fsm has already applied a later snapshot (from remote) that we didn't yet 'observe'
-               // (i.e. didn't yet receive from _apply_entries queue) but will soon. We avoid unnecessary work
-               // of taking snapshots ourselves but comparing our last index directly with what's currently in _fsm.
-               auto last_snap_idx = _fsm->log_last_snapshot_idx();
-
-               // Error injection to be set with one_shot
-               utils::get_local_injector().inject("raft_server_snapshot_reduce_threshold",
-                   [this] { _config.snapshot_threshold = 3; _config.snapshot_trailing = 1; });
-
-               co_await override_snapshot_thresholds();
-
-               bool force_snapshot = utils::get_local_injector().enter("raft_server_force_snapshot");
-
-               if (force_snapshot || (_applied_idx > last_snap_idx &&
-                   ((_applied_idx - last_snap_idx).value() >= _config.snapshot_threshold ||
-                   _fsm->log_memory_usage() >= _config.snapshot_threshold_log_size)))
-               {
-                   snapshot_descriptor snp;
-                   snp.term = last_term;
-                   snp.idx = _applied_idx;
-                   snp.config = _fsm->log_last_conf_for(_applied_idx);
-                   logger.trace("[{}] applier fiber: taking snapshot term={}, idx={}", _id, snp.term, snp.idx);
-                   snp.id = co_await _state_machine->take_snapshot();
-                   // Note that at this point (after the `co_await`), _fsm may already have applied a later snapshot.
-                   // That's fine, `_fsm->apply_snapshot` will simply ignore our current attempt; we will soon receive
-                   // a later snapshot from the queue.
-                   auto max_trailing = force_snapshot ? 0 : _config.snapshot_trailing;
-                   auto max_trailing_bytes = force_snapshot ? 0 : _config.snapshot_trailing_size;
-                   if (!_fsm->apply_snapshot(snp, max_trailing, max_trailing_bytes, true)) {
-                       logger.trace("[{}] applier fiber: while taking snapshot term={} idx={} id={},"
-                              " fsm received a later snapshot at idx={}", _id, snp.term, snp.idx, snp.id, _fsm->log_last_snapshot_idx());
-                   }
-                   _stats.snapshots_taken++;
-               }
+                _applied_idx = last_idx;
+                _applied_index_changed.broadcast();
+                notify_waiters(_awaited_applies, batch);
+
+                // It may happen that _fsm has already applied a later snapshot (from remote) that we didn't yet 'observe'
+                // (i.e. didn't yet receive from _apply_entries queue) but will soon. We avoid unnecessary work
+                // of taking snapshots ourselves but comparing our last index directly with what's currently in _fsm.
+                const auto last_snap_idx = _fsm->log_last_snapshot_idx();
```

Comment:
A wild `const` has appeared in a commit titled `Fix indentation`. That's cheating!

Reply:
Sorry, it was a reflex.